macros/src/attr/struct.rs

```diff
@@ -28,67 +28,88 @@ pub struct SerdeStructAttr(StructAttr);
 
 impl StructAttr {
     pub fn from_attrs(attrs: &[Attribute]) -> Result<Self> {
-        let mut result = Self::default();
-        parse_attrs(attrs)?.for_each(|a| result.merge(a));
+        let mut result = parse_attrs(attrs)?.fold(Self::default(), |acc, cur| acc.merge(cur));
 
         let docs = parse_docs(attrs)?;
         result.docs = docs;
 
         #[cfg(feature = "serde-compat")]
-        crate::utils::parse_serde_attrs::<SerdeStructAttr>(attrs).for_each(|a| result.merge(a.0));
+        let result = crate::utils::parse_serde_attrs::<SerdeStructAttr>(attrs)
+            .fold(result, |acc, cur| acc.merge(cur.0));
         Ok(result)
     }
 
-    pub fn from_variant(enum_attr: &EnumAttr, variant_attr: &VariantAttr) -> Self {
+    pub fn from_variant(
+        enum_attr: &EnumAttr,
+        variant_attr: &VariantAttr,
+        variant_fields: &Fields,
+    ) -> Self {
         Self {
             crate_rename: Some(enum_attr.crate_rename()),
             rename: variant_attr.rename.clone(),
-            rename_all: variant_attr.rename_all,
+            rename_all: variant_attr.rename_all.or(match variant_fields {
+                Fields::Named(_) => enum_attr.rename_all_fields,
+                Fields::Unnamed(_) | Fields::Unit => None,
+            }),
```

Comment:
I managed to move the `rename_all_fields` match to a better location and get rid of that duplicated mehtod